### PR TITLE
Verify auth configs

### PR DESCRIPTION
**Pull request by ashleywang1**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2623

### CODE REVIEW

File: projects/gloo/pkg/translator/authconfigs.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		}
+		for _, conf := range configs {
+			switch cfg := conf.AuthConfig.(type) {
+			case *extauthv1.AuthConfig_Config_BasicAuth:
+				if cfg.BasicAuth.GetApr() == nil || cfg.BasicAuth.GetRealm() == "" {
+					reports.AddError(ac, errors.Errorf("Invalid configurations for basic auth config %v", ac.Metadata.String()))
+				}
+			case *extauthv1.AuthConfig_Config_Oauth:
+				if cfg.Oauth.GetAppUrl() == "" {
+					reports.AddError(ac, errors.Errorf("Invalid configurations for basic auth config %v", ac.Metadata.String()))
```
Comment (kdorosh) [inline projects/gloo/pkg/translator/authconfigs.go:25]:
```suggestion
					reports.AddError(ac, errors.Errorf("Invalid configurations for oauth config %v", ac.Metadata.String()))
```

File: projects/gloo/pkg/translator/authconfigs.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+				}
+			case *extauthv1.AuthConfig_Config_ApiKeyAuth:
+				if len(cfg.ApiKeyAuth.GetLabelSelector())+len(cfg.ApiKeyAuth.GetLabelSelector()) == 0 {
```
Comment (kdorosh) [inline projects/gloo/pkg/translator/authconfigs.go:28]: why do we add the sum of the length twice here?

File: projects/gloo/pkg/translator/authconfigs.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+				if cfg.PluginAuth.GetConfig() == nil {
+					reports.AddError(ac, errors.Errorf("Invalid configurations for plugin auth config %v", ac.Metadata.String()))
+				}
+			case *extauthv1.AuthConfig_Config_OpaAuth: // no way to verify this from just the config
+			case *extauthv1.AuthConfig_Config_Ldap:
+				if cfg.Ldap.GetAddress() == "" {
+					reports.AddError(ac, errors.Errorf("Invalid configurations for ldap auth config %v", ac.Metadata.String()))
+				}
+			default:
+				reports.AddError(ac, errors.Errorf("Unknown Auth Config type for %v", ac.Metadata.String()))
```
Comment (kdorosh) [inline projects/gloo/pkg/translator/authconfigs.go:41]: `ac.Metadata.String()` is pretty verbose, perhaps we want to be using `ac.Metadata.Ref().String()`

Comment (kdorosh) [reply projects/gloo/pkg/translator/authconfigs.go:41]: to improve the readability of https://github.com/solo-io/gloo/issues/2623#issuecomment-607429053

File: projects/gloo/pkg/translator/authconfigs.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	authConfigs := params.Snapshot.AuthConfigs
+	for _, ac := range authConfigs {
+		configs := ac.GetConfigs()
+		if len(configs) == 0 {
+			reports.AddError(ac, errors.Errorf("No configurations for auth config %v", ac.Metadata.String()))
+		}
+		for _, conf := range configs {
+			switch cfg := conf.AuthConfig.(type) {
+			case *extauthv1.AuthConfig_Config_BasicAuth:
+				if cfg.BasicAuth.GetApr() == nil || cfg.BasicAuth.GetRealm() == "" {
```
Comment (kdorosh) [inline projects/gloo/pkg/translator/authconfigs.go:20]: realm can be omitted or empty, in fact our documented example has it empty: https://docs.solo.io/gloo/latest/guides/security/auth/basic_auth/

File: projects/gloo/pkg/translator/authconfigs.go (hunk 2 above)
Comment (ashleywang1) [reply projects/gloo/pkg/translator/authconfigs.go:28]: oh whoops! It's supposed to be GetApiKeySecretRefs
We need to have at least one of the secret refs or the label selectors. That's what this was checking for.

File: projects/gloo/pkg/translator/authconfigs.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+				}
+			case *extauthv1.AuthConfig_Config_ApiKeyAuth:
+				if len(cfg.ApiKeyAuth.GetLabelSelector())+len(cfg.ApiKeyAuth.GetLabelSelector()) == 0 {
+					reports.AddError(ac, errors.Errorf("Invalid configurations for apikey auth config %v", ac.Metadata.String()))
+				}
+			case *extauthv1.AuthConfig_Config_PluginAuth:
+				if cfg.PluginAuth.GetConfig() == nil {
+					reports.AddError(ac, errors.Errorf("Invalid configurations for plugin auth config %v", ac.Metadata.String()))
+				}
+			case *extauthv1.AuthConfig_Config_OpaAuth: // no way to verify this from just the config
```
Comment (kdorosh) [inline projects/gloo/pkg/translator/authconfigs.go:35]: we should be able to check that the query isn't empty

**Comment by ashleywang1** [comment]
Moving this code to the extauthPlugin code in Gloo Enterprise, since we should only be validating auth configs when we're running GlooE

**Pull request closed without merging**